bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/CTabRendering.java

```diff
@@ -624,7 +624,7 @@ void drawSelectedTab(int itemIndex, GC gc, Rectangle bounds) {
 			int verticalOffset = highlightOnTop ? 0 : bounds.height - 2;
```

Comment:
I think the cause for the misalignment is the combination with this magic number used for the vertical placement. Changing it to `1` seems to solve the issue.

I propose to put the highlight thickness into a separate variable and use it here for the offset calculation as well as for the line thickness. That improves modifiability and comprehensibility (like always when facing magic number 🙂 ).